### PR TITLE
fix: remove bc optional dependency

### PR DESCRIPTION
**Pull request by shawkins**
## Description

Removes the bc optional dependency

closes: #6008

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/main/CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/main/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@coopstah13 : Could you please review it to see if it meets your expectations?

**Comment by shawkins** [comment]
@rohanKanojia unfortunately it still does have an issue. The remaining failures are related to: 
```
java.security.InvalidAlgorithmParameterException: Curve not supported: secp256k1 (1.3.132.0.10)
```
JDK 15+ dropped support for secp256k1, but we're using a key from that algorithm for KubernetesClientLoadWithFipsProviderTest.

We can either:
- Bring back a little bit of our reliance on bouncy castle - however we don't need an optional dependency. If the user installs bc / bcfips as a provider this will work if we instead call KeyPairGenerator.getInstance("EC", bc_provider)
- Drop support for this EC - do we know what is in use for supported versions of kubernetes?

**Comment by coopstah13** [comment]
I think it would be acceptable to leave fall-back code with the optional dependency for supporting that. Basically, first you'd try to do it using JDK internals, but maybe catch the exception and fall back to the Callable code which lets it dynamically link to the BC dependency and leave BC as an optional dependency defined in your pom.

Appreciate the effort on this.

**Comment by shawkins** [comment]
I've opted for suggesting that BC be installed as a provider if there's a problem with the algorithm. This approach means that we won't have any direct compile / runtime dependency on BC.

File: kubernetes-client-api/pom.xml [hunk 1]
```diff
@@ -160,12 +160,10 @@
     <dependency>
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcprov-jdk18on</artifactId>
-      <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcpkix-jdk18on</artifactId>
-      <optional>true</optional>
```
Comment (coopstah13) [inline kubernetes-client-api/pom.xml:168]: this change seems odd to me

Comment (shawkins) [reply kubernetes-client-api/pom.xml:168]: Why? It's now a test dependency.

**Review by manusa — APPROVED** [review]
LGTM, nice job.
Thx!

**Pull request merged**